### PR TITLE
Implementation of TimeStamps for events #56

### PR DESCRIPTION
**Pull request by Ayybeeshafi**
Last branch had come conflicts with the develop branch. This one is up to date.

### CODE REVIEW

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala [hunk 1]
```diff
@@ -79,9 +80,21 @@ trait EventJsonSupport extends SprayJsonSupport with DefaultJsonProtocol with In
     }
 
   }
+  implicit val timestamp: JsonFormat[DateTime] = new JsonFormat[DateTime] {
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala]: Please rename this val to ```timestampFormat``` to be consistent with the other formats. Also please add JavaDoc comments to this val.

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala [hunk 2]
```diff
@@ -79,9 +80,21 @@ trait EventJsonSupport extends SprayJsonSupport with DefaultJsonProtocol with In
...
+    override def write(obj: DateTime) = JsString(obj.toIsoDateTimeString())
+
+    override def read(json: JsValue): DateTime = json match {
+      case JsString(value) =>
+        DateTime.fromIsoDateTimeString(value) match {
+          case Some(date) => date
+          case _ => throw new DeserializationException("Failed to parse date time [" + value + "].")
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala]: Please remove the ```new``` keyword here, it is not necessary and marked as warning by IntelliJ.

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala [hunk 3]
```diff
@@ -79,9 +80,21 @@ trait EventJsonSupport extends SprayJsonSupport with DefaultJsonProtocol with In
...
+        }
+      case _ => throw new DeserializationException("Failed to parse json string [" + json + "].")
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala]: Please remove the ```new``` keyword here, it is not necessary and marked as warning by IntelliJ.

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala [hunk 4]
```diff
@@ -121,31 +136,31 @@ object RegistryEventFactory {
     * @return RegistryEvent with the respective type and payload.
     */
   def createNumbersChangedEvent(componentType: ComponentType, newNumber: Int) : RegistryEvent =
-    RegistryEvent(EventType.NumbersChangedEvent, NumbersChangedPayload(componentType, newNumber))
+    RegistryEvent(EventType.NumbersChangedEvent, NumbersChangedPayload(componentType, newNumber),DateTime.now)
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/io/swagger/client/model/Event.scala]: Please add a space after the comma. This also holds for the other occurrences below.

**Review by johannesduesing — CHANGES_REQUESTED** [review]
Thanks for your PR :+1: . Please address the codestyle issues mentioned in my comments below.

**Comment by Ayybeeshafi** [comment]
@johannesduesing Thankyou for the review. Please verify the requested changes in the latest commit d4ce17e

**Review by johannesduesing — APPROVED** [review]
Thanks for the quick corrections :+1:

**Pull request merged**